maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -1399,12 +1400,22 @@ private static void checkThreadCountEntity( int count, String entity )
 
     private boolean isJunit47Compatible( Artifact artifact )
     {
-        return dependencyResolver.isWithinVersionSpec( artifact, "[4.7,)" );
+        return dependencyResolver.isWithinVersionSpec( artifact, "[4.7,4.12.0)" );
```

Comment:
JUnit 5 has another groupId so there is no reason to limit the version. JUnit 5 is just another artifact.